src/core/lib/event_engine/grpc_ares_wrapper.cc

```diff
@@ -0,0 +1,1025 @@
+    return;
+  }
+  // Find service config in TXT record.
+  struct ares_txt_ext* result = nullptr;
+  for (result = reply; result != nullptr; result = result->next) {
+    if (result->record_start &&
+        memcmp(result->txt, g_service_config_attribute_prefix, g_prefix_len) ==
+            0) {
+      break;
+    }
+  }
+  std::string service_config_json_out;
+  // Found a service config record.
+  if (result != nullptr) {
+    size_t service_config_len = result->length - g_prefix_len;
+    service_config_json_out.append(
+        reinterpret_cast<char*>(result->txt) + g_prefix_len,
+        service_config_len);
+    for (result = result->next; result != nullptr && !result->record_start;
+         result = result->next) {
+      service_config_json_out.append(reinterpret_cast<char*>(result->txt),
+                                     result->length);
+    }
+    GRPC_ARES_WRAPPER_TRACE_LOG("request:%p found service config: %s", r,
+                                service_config_json_out.c_str());
+  }
+  // Clean up.
+  ares_free_data(reply);
+  r->OnResolveLocked(std::move(service_config_json_out));
+}
+
+GrpcAresRequest::FdNode::FdNode(ares_socket_t as, GrpcPolledFd* polled_fd)
+    : as(as), polled_fd(polled_fd) {}
+
+GrpcAresRequest::FdNode* GrpcAresRequest::FdNodeList::PopFdNode(
+    ares_socket_t as) {
+  FdNode phony_head;
+  phony_head.next = head_;
+  FdNode* node = &phony_head;
+  while (node->next != nullptr) {
+    if (node->next->polled_fd->GetWrappedAresSocketLocked() == as) {
+      FdNode* ret = node->next;
+      node->next = node->next->next;
+      head_ = phony_head.next;
+      return ret;
+    }
+    node = node->next;
+  }
+  return nullptr;
+}
+
+GrpcAresRequest::GrpcAresRequest(
+    absl::string_view name, EventEngine::Duration timeout,
+    std::unique_ptr<GrpcPolledFdFactory> polled_fd_factory,
+    EventEngine* event_engine)
+    : grpc_core::InternallyRefCounted<GrpcAresRequest>(
+          GRPC_TRACE_FLAG_ENABLED(grpc_trace_ares_wrapper) ? "GrpcAresRequest"
+                                                           : nullptr),
+      name_(name),
+      event_engine_(event_engine),
+      timeout_(timeout),
+      fd_node_list_(std::make_unique<FdNodeList>()),
+      polled_fd_factory_(std::move(polled_fd_factory)) {}
+
+GrpcAresRequest::~GrpcAresRequest() {
+  if (initialized_) {
+    ares_destroy(channel_);
+    PrintCurrentStackTrace();
+  }
+  GRPC_ARES_WRAPPER_TRACE_LOG("request:%p destructor", this);
+}
+
+absl::StatusOr<std::string> GrpcAresRequest::ParseNameToResolve()
+    ABSL_NO_THREAD_SAFETY_ANALYSIS {
+  GPR_DEBUG_ASSERT(!initialized_);
+  std::string port;
+  // parse name, splitting it into host and port parts
+  grpc_core::SplitHostPort(name_, &host_, &port);
+  absl::Status error;
+  if (host_.empty()) {
+    error =
+        grpc_error_set_str(GRPC_ERROR_CREATE("unparseable host:port"),
+                           grpc_core::StatusStrProperty::kTargetAddress, name_);
+    return error;
```

Comment:
nit: this can just be `return grpc_error_set_str(...);`

Reply:
Done.